arangod/Aql/Function/ObjectFunctions.cpp

```diff
@@ -774,4 +774,39 @@ AqlValue functions::Zip(ExpressionContext* expressionContext, AstNode const&,
   return AqlValue(builder->slice(), builder->size());
 }
 
+AqlValue functions::Entries(ExpressionContext* expressionContext,
+                            AstNode const&,
+                            VPackFunctionParametersView parameters) {
+  // cppcheck-suppress variableScope
+  static char const* AFN = "ENTRIES";
+
+  AqlValue const& object =
+      aql::functions::extractFunctionParameterValue(parameters, 0);
+
+  if (!object.isObject()) {
+    registerWarning(expressionContext, AFN,
+                    TRI_ERROR_QUERY_FUNCTION_ARGUMENT_TYPE_MISMATCH);
+    return AqlValue(AqlValueHintNull());
+  }
+
+  transaction::Methods* trx = &expressionContext->trx();
+  auto* vopts = &trx->vpackOptions();
+
+  AqlValueMaterializer objectMaterializer(vopts);
+  VPackSlice objectSlice = objectMaterializer.slice(object);
+
+  transaction::BuilderLeaser builder(trx);
+  builder->openArray();
+
+  for (auto [key, value] : VPackObjectIterator(objectSlice)) {
```

Comment:
I strongly suggest that we iterate over the attributes of the object in _object build order_ here. This is indicated by passing the `useSequentialIteration = true` flag to the velocypack::ObjectIterator. This allows a linear scan over the object's underlying memory.
Not passing this flag means we will be iterating over the attributes in _index table order_ here. This will mean random accesses to the object entries for each index table entry, which can be a lot less efficient for larger objects or objects with many keys.

```suggestion
  for (auto [key, value] : VPackObjectIterator(objectSlice, true)) {
```

Reply:
Valid point. I will fix that.